File: 167_TwoSumII-InputArraySorted/167_TwoSumII-InputArraySorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let left = 0,
    right = numbers.length - 1;
  while (left < right) {
    const compare = numbers[right] + numbers[left];
    if (target == compare) return [left + 1, right + 1];
    else if (target > compare) left++;
    else right--;
  }
};

File: 167_TwoSumII-InputArraySorted/167_TwoSumII-InputArraySorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        while left < right:
            compare = numbers[left] + numbers[right]
            if target < compare:
                right -= 1
            elif target > compare:
                left += 1
            else:
                return [left + 1, right + 1]
    
            
